org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/launch/BaseLaunchRequestJob.java

```diff
@@ -70,12 +72,23 @@ protected final void runToolingApiJob(IProgressMonitor monitor) {
         // fetch build environment
         List<ProgressListener> listeners = ImmutableList.<ProgressListener>of(DelegatingProgressListener.withFullOutput(monitor));
 
-        // apply the fixed attributes on the request o
         BuildRequest<Void> request = createRequest();
-        FixedRequestAttributes fixedAttributes = getConfigurationAttributes().toFixedRequestAttributes();
-        fixedAttributes.apply(request);
-
-        // configure the request's transient attributes
+        RunConfiguration runConfig = getRunConfig();
+        BuildConfiguration buildConfig = runConfig.getBuildConfiguration();
+        // TODO (donat) this won't be necessary once the project config can override Gradle user home
+        WorkspaceConfiguration workspaceConfig = CorePlugin.configurationManager().loadWorkspaceConfiguration();
+
+        request.gradleDistribution(buildConfig.getGradleDistribution());
+
+        // apply the fixed attributes
+        request.projectDir(buildConfig.getRootProjectDirectory());
+        request.gradleUserHomeDir(workspaceConfig.getGradleUserHome());
+        request.gradleDistribution(buildConfig.getGradleDistribution());
+        request.javaHomeDir(runConfig.getJavaHome());
+        request.jvmArguments(runConfig.getJvmArguments().toArray(new String[0]));
+        request.arguments(runConfig.getArguments().toArray(new String[0]));
```

Comment:
The run config should have an `applyTo(Request)`  method which handles setting all the properties on the request. It should delegate to a similar apply method on its build config, which in turn delegates first to the workspace config.